src/rp2_common/hardware_clocks/clocks.c

```diff
@@ -245,10 +245,20 @@ void clock_gpio_init_int_frac(uint gpio, uint src, uint32_t div_int, uint8_t div
         invalid_params_if(HARDWARE_CLOCKS, true);
     }
 
+#if !PICO_RP2040 // assert currently broken on RP2040, but we know that hardware has 16-bit integer part
```

Comment:
Clock-divider hardware on RP2040 has 24-bit integer part

Reply:
reworked the assert